src/main.cpp

```diff
@@ -768,9 +811,9 @@ int main ( int argc, char** argv )
     bIsClient      = true; // Client only - TODO: maybe a switch in interface to change to server?
 
     // bUseMultithreading = true;
-    QApplication* pApp = new QApplication ( argc, argv );
+    pApp = new QApplication ( argc, argv );
```

Comment:
So, for iOS apps this used to be of type `QApplication` instead of `QCoreApplication`. Can you give a quick explanation why this is safe to do?

It's probably hard or artificial to put that into an individual commit, but at least the commit description should mention that this was intentional and why it is considered safe (I guess because iOS doesn't use any methods from the broader interface either?).

Reply:
???

I didn't change the type, I just moved the pointer declaration outside main() to be able to implement the cleanup() function.

Reply:
Yeah, I understand the reason for moving. I just wanted to hint at the side effect (maybe it's a noop, but I'm not sure):
`pApp` was previously either of type `QApplication` or `QCoreApplication` based on platform and/or GUI/non-GUI.
With the new code, it's always a `QCoreApplication` which is probably the lowest common denominator. It might be fine, but it's still a change for iOS, isn't it?

Reply:
I'm still wondering why you think anything changed for IOS, AFAIK It's was, and still is, always QApplication for IOS.

```
#    if defined( Q_OS_IOS )
    bUseGUI        = true;
    bIsClient      = true; // Client only - TODO: maybe a switch in interface to change to server?

    // bUseMultithreading = true;
    pApp = new QApplication ( argc, argv );
#    else

```

Reply:
OK, Now I see your point. Thanks for noting!

QCoreApplication is the base type of all QApplications, that's why I can use that type to store the pointer. 
But I expected yhe basetype to have virtual functions, but now I checked it and that isn't the case... 
Though all QApplications will call QCoreApplication::exec() eventually there might be a difference in behavior, so I will have to find a better solution for this.